### PR TITLE
fix: Rename Module to Plugin everywhere

### PR DESCRIPTION
**Pull request by danielledeleo**

### CODE REVIEW

File: tests/hello.ts [hunk 1]
```diff
@@ -1,9 +0,0 @@
-import { logInfo } from "@suborbital/suborbital"
```
Comment (rnpridgeon) [inline tests/hello.ts:1]: Is there a reason for removing the test?

File: README.md [hunk 2]
```diff
@@ -22,23 +22,23 @@ import (
...
 
     client, err := se2.NewClient(se2.LocalConfig(), token)
     if err != nil {
         log.Fatal(err)
     }
 
-    // get a list of Modules
-    modules, err := client.UserFunctions("userID", "namespace")
+    // get a list of Plugins
+    plugins, err := client.UserFunctions("userID", "namespace")
```
Comment (ospencer) [inline README.md]: Should be `UserPlugins`

File: examples/app/app.go [hunk 3]
```diff
@@ -1,38 +1,53 @@
...
 
 	se2 "github.com/suborbital/se2-go"
 )
 
 func main() {
 	client := client()
 
-	// This is a local reference to some Module. Nothing has run in Compute at this point.
-	module := se2.NewModule("com.suborbital", "acmeco", "default", "rs-hello-world")
+	// This is a local reference to some Plugin. Nothing has run in Compute at this point.
```
Comment (ospencer) [inline examples/app/app.go]: SE2

File: examples/app/app.go [hunk 4]
```diff
@@ -1,38 +1,53 @@
...
-	// Deploy the function (the module's .Version field is adjusted here)
-	client.PromoteDraft(module)
+	// Deploy the function and get the new reference
+	ref, _ := client.PromoteDraft(plugin)
 
-	// Execute the function
-	result, _, _ := client.ExecString(module, "world!")
+	// Roll the credits
+	time.Sleep(time.Second * 2)
+	for _, name := range []string{"Connor", "Dan", "Dylan", "Flaki", "Gabor", "Jagger", "Jessica", "Laura", "Nyah", "Oscar", "Ramón", "Ryan", "Taryn"} {
```
Comment (ospencer) [inline examples/app/app.go]: I love this but I think it's going to be a pain to keep updated :)

File: examples/basic/main.go [hunk 5]
```diff
@@ -21,30 +21,30 @@ func main() {
...
 	if err != nil {
 		log.Panic(err)
 	}
 
 	log.Printf("building with template:\n%s\n", template.Contents)
 
-	// trigger a remote build of the module
-	buildResult, _ := client.BuildFunctionString(helloModule, tmpl, template.Contents)
+	// trigger a remote build of the plugin
+	buildResult, _ := client.BuildFunctionString(helloPlugin, tmpl, template.Contents)
```
Comment (ospencer) [inline examples/basic/main.go]: should be `BuildPluginString`

File: tests/hello.ts (hunk 1 above)
Comment (danielledeleo) [reply tests/hello.ts:1]: It's not a Go test and it wasn't referenced everywhere as far as I can tell.

File: README.md [hunk 6]
```diff
@@ -10,7 +10,7 @@ In a Go project, run
 go get github.com/suborbital/se2-go@latest
 ```
 
-Every operation with Compute is done with a `se2.Client`. Here's a simple example that fetches exisiting Modules for a user and namespace.
+Every operation with SE2 is done with a `se2.Client`. Here's a simple example that fetches exisiting Plugins for a user and namespace.
```
Comment (arbourd) [inline README.md]:
```suggestion
Every operation with SE2 is done with a `se2.Client`. Here's a simple example that fetches existing plugins for a user and namespace.
```

Nit, just based on some other feedback elsewhere.

File: builder_requests.go [hunk 7]
```diff
@@ -52,21 +52,21 @@ func (c *Client) BuilderFeatures() (*FeaturesResponse, error) {
 	return features, nil
 }
 
-// BuilderTemplate gets the function template for the provided Module and template name.
-func (c *Client) BuilderTemplate(module *Module, template string) (*EditorStateResponse, error) {
-	if module == nil {
-		return nil, errors.New("Module cannot be nil")
+// BuilderTemplate gets the plugin template for the provided Plugin and template name.
```
Comment (arbourd) [inline builder_requests.go]:
```suggestion
// BuilderTemplate gets the plugin template for the provided plugin and template name.
```

Nit based on feedback in other repos.

File: builder_requests.go [hunk 8]
```diff
@@ -86,32 +86,32 @@ func (c *Client) BuilderTemplate(module *Module, template string) (*EditorStateR
 	return editorState, nil
 }
 
-// BuildFunction triggers a remote build for the given Module and function body. See also: Client.BuildFunctionString()
+// BuildPlugin triggers a remote build for the given Plugin and source code. See also: Client.BuildPluginString()
```
Comment (arbourd) [inline builder_requests.go]:
```suggestion
// BuildPlugin triggers a remote build for the given plugin and source code. See also: Client.BuildPluginString()
```
Nit based on feedback in other repos.

File: builder_requests.go [hunk 9]
```diff
@@ -135,25 +135,25 @@ func (c *Client) BuildFunction(module *Module, template string, functionBody io.
 	return buildResult, nil
 }
 
-// BuildFunctionString triggers a remote build for the given Module and function string. See also: Client.BuildFunction()
-func (c *Client) BuildFunctionString(module *Module, template, functionString string) (*BuildResult, error) {
-	buf := bytes.NewBufferString(functionString)
-	return c.BuildFunction(module, template, buf)
+// BuildPluginString triggers a remote build for the given Plugin and source code. See also: Client.BuildPlugin()
```
Comment (arbourd) [inline builder_requests.go]:
```suggestion
// BuildPluginString triggers a remote build for the given plugin and source code. See also: Client.BuildPlugin()
```
Nit based on feedback in other repos.

File: builder_requests.go [hunk 10]
```diff
@@ -135,25 +135,25 @@ func (c *Client) BuildFunction(module *Module, template string, functionBody io.
...
+func (c *Client) BuildPluginString(plugin *Plugin, template, source string) (*BuildResult, error) {
+	buf := bytes.NewBufferString(source)
+	return c.BuildPlugin(plugin, template, buf)
 }
 
-// GetDraft gets the most recently build source code for the provided Module. Must have the .FQFNURI field set.
-func (c *Client) GetDraft(module *Module) (*EditorStateResponse, error) {
-	if module == nil {
-		return nil, errors.New("Module cannot be nil")
+// GetDraft gets the most recently build source code for the provided Plugin. Must have the .FQFNURI field set.
```
Comment (arbourd) [inline builder_requests.go]:
```suggestion
// GetDraft gets the most recently build source code for the provided plugin. Must have the .FQMNURI field set.
```
Nit based on feedback in other repos.

File: examples/getplugins/main.go [hunk 11]
```diff
@@ -18,13 +18,13 @@ func main() {
 		log.Fatal(err)
 	}
 
-	// get a list of Modules
-	modules, err := client.UserFunctions("userID", "namespace")
+	// get a list of Plugins
```
Comment (arbourd) [inline examples/getplugins/main.go]:
```suggestion
	// get a list of plugins
```
Nit based on feedback in other repos.

File: plugin.go [hunk 12]
```diff
@@ -2,26 +2,26 @@ package se2
...
 	Tenant      string
 	Namespace   string
 	Name        string
 }
 
-// NewModule instantiates a local v1.0.0 Module that can be used for various calls with compute.Client.
-// Note: this constructor alone does not perform any actions on a remote Compute instance.
-func NewModule(environment, tenant, namespace, name string) *Module {
-	module := &Module{
+// NewPlugin instantiates a local v1.0.0 Plugin that can be used for various calls with se2.Client.
```
Comment (arbourd) [inline plugin.go]:
```suggestion
// NewPlugin instantiates a local v1.0.0 plugin that can be used for various calls with se2.Client.
```
Nit based on feedback in other repos.

File: response_types.go [hunk 13]
```diff
@@ -9,8 +9,8 @@ import (
 type TokenResponse struct {
 	Token string `json:"token"`
 }
-type UserFunctionsResponse struct {
-	Functions []*tenant.Module `json:"functions"`
+type UserPluginsResponse struct {
+	Plugins []*tenant.Module `json:"modules"`
```
Comment (arbourd) [inline response_types.go:13]: Can we change this now?

File: response_types.go [hunk 14]
```diff
@@ -36,7 +36,7 @@ type FeaturesResponse struct {
 	Langauges []FeatureLanguage `json:"languages"`
 }
 
-// TestPayload is a single test for a Module
+// TestPayload is a single test for a Plugin
```
Comment (arbourd) [inline response_types.go]:
```suggestion
// TestPayload is a single test for a plugin
```
Nit based on feedback in other repos.

File: se2_test.go [hunk 15]
```diff
@@ -163,7 +164,7 @@ func TestGetAndExec(t *testing.T) {
 
 	// Tests the administrative results endpoint
 	t.Run("ExecResultMetadata", func(t *testing.T) {
-		res, err := client.FunctionResultMetadata(uuid)
+		res, err := client.ResultMetadata(uuid)
```
Comment (arbourd) [inline se2_test.go]: `ResultMetadata` or `ExecutionResultMetadata`?

We should match with the node one as well.

File: response_types.go (hunk 13 above)
Comment (danielledeleo) [reply response_types.go:13]: Yes this is already merged into `se2`

**Review by arbourd — APPROVED** [review]

**Pull request merged**